### PR TITLE
remove linear_chain_crf and crf_decoding from fluid

### PR DESCRIPTION
**Pull request by ccrrong**
### PR types

Others

### PR changes

APIs

### Describe

To clean the code in paddle.fluid, the useless codes should be removed and code still be used should move to a new position.

For function paddle.fluid.layers.nn.linear_chain_crf and paddle.fluid.layers.nn.crf_decoding, since it is not used in version 2.x, should be removed.

### CODE REVIEW

**Review by heavengate — APPROVED** [review]

**Review by zhwesky2010 — APPROVED** [review]

File: python/paddle/fluid/tests/book/test_label_semantic_roles.py [hunk 1]
```diff
@@ -162,12 +162,8 @@ def train(use_cuda, save_dirname=None, is_local=True):
     target = fluid.layers.data(
         name='target', shape=[1], dtype='int64', lod_level=1
     )
-    crf_cost = fluid.layers.linear_chain_crf(
-        input=feature_out,
-        label=target,
-        param_attr=fluid.ParamAttr(name='crfw', learning_rate=mix_hidden_lr),
-    )
-    avg_cost = paddle.mean(crf_cost)
+    cost = fluid.layers.softmax_with_cross_entropy(feature_out, target)
```
Comment (XiaoguangHu01) [inline python/paddle/fluid/tests/book/test_label_semantic_roles.py:165]: 这里加这个单测的原因是什么？

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by Ligoml — APPROVED** [review]
LGTM for docs

**Review by jzhang533 — APPROVED** [review]
LGTM for fluid API removing

**Review by heavengate — APPROVED** [review]

**Pull request merged**